Frameworks/Foundation/NSOperation.mm

```diff
@@ -131,139 +166,123 @@ - (void)setThreadPriority:(double)priority {
  @Status Interoperable
 */
 - (void)setCompletionBlock:(void (^)(void))block {
-    id oldBlock = priv->completionBlock;
-    priv->completionBlock = [block copy];
-    [oldBlock release];
+    std::lock_guard<std::recursive_mutex> lock(_completionBlockLock);
+    [self willChangeValueForKey:@"completionBlock"];
+    _completionBlock = block;
+    [self didChangeValueForKey:@"completionBlock"];
 }
 
 /**
  @Status Interoperable
 */
 - (void (^)(void))completionBlock {
-    return priv->completionBlock;
+    std::lock_guard<std::recursive_mutex> lock(_completionBlockLock);
+    return [[_completionBlock retain] autorelease];
 }
 
 /**
  @Status Interoperable
 */
-- (BOOL)isReady {
-    //  Note, check dependencies when we get them
-    int count = [priv->dependencies count];
-
-    for (int i = 0; i < count; i++) {
-        id op = [priv->dependencies objectAtIndex:i];
-
-        if (![op isFinished]) {
-            return NO;
-        }
-    }
-    return YES;
+- (BOOL)isCancelled {
+    std::lock_guard<std::recursive_mutex> lock(_finishLock);
+    return _cancelled;
```

Comment:
I don't think the lock is needed for the accessor. Removing.

---
In reply to: [69818734](https://github.com/Microsoft/WinObjC/pull/602#discussion_r69818734) [](ancestors = 69818734)